src/common/libflux/flog.c

```diff
@@ -111,6 +111,31 @@ static int log_rpc (flux_t *h, const char *buf, size_t len)
     return 0;
 }
 
+static int vlog_stderr (int level, const char *fmt, va_list ap)
+{
+    char buf[FLUX_MAX_LOGBUF];
+    char *result = buf;
+    char *mbuf = NULL;
+    const char *lstr = stdlog_severity_to_string (LOG_PRI (level));
+    int rc;
+    va_list ap_copy;
+
+    /* Copy va_list ap in case it reused on truncation below:
+     */
+    va_copy (ap_copy, ap);
+
+    if (vsnprintf (result, sizeof (buf), fmt, ap) >= sizeof (buf)) {
+        /* Message truncation occurred. Fall back to malloc. If that
+         * fails, just print the truncated message:
+         */
+        if (vasprintf (&mbuf, fmt, ap_copy) == 0)
+            result = mbuf;
+    }
+    rc = fprintf (stderr, "%s: %s\n", lstr, result);
+    free (mbuf);
+    return rc < 0 ? -1 : 0;
```

Comment:
va_copy(3) says
> Each  invocation of va_copy() must be matched by a corresponding invocation
       of va_end() in  the  same  function. 

Add one after the conditional block?

Reply:
D'oh! Thanks for catching this one!